utils/ctranslate.py

```diff
@@ -0,0 +1,62 @@
+import csv
+import json
+import os
+import time
+
+import ctranslate2
+import sentencepiece
+
+import utils.benchmark as bench_utils
+from utils.misc import advertise_aio
+
+class CTranslateRunner:
+    """
+    A class providing facilities to run CTranslate2 model.
+    """
+
+    def __init__(self, model, tokenizer):
+        # try:
+        #     #TODO: Check for AIO
+        # except AttributeError:
+        #     utils.advertise_aio("CTranslate2")
+
+        self.translator = ctranslate2.Translator(model, device='cpu', compute_type='auto', inter_threads=1, intra_threads=bench_utils.get_intra_op_parallelism_threads())
```

Comment:
Line 186 here https://github.com/OpenNMT/CTranslate2/blob/c84a826bec6278c36a776f9a32da4fdc3f76f421/src/types.cc sheds some light on compute_type "auto" behaviour. I would say explicitly asking for the requested dtype would be safer.